### PR TITLE
Refactor for code reuse

### PR DESCRIPTION
**Pull request by servingUpAces**
- Refactor send_sms and send_email to use common code to persist the notificaiton.
- Use the same method to persist a notification.
- Refactored the send_sms task to use this method.
- Fix the send_sms and send_email code to send the notification id that has been persisted.
- Fix logging in send_sms and send_email

### CODE REVIEW

File: tests/app/celery/test_tasks.py [hunk 1]
```diff
@@ -337,24 +330,17 @@ def test_should_send_template_to_correct_sms_task_and_persist(sample_template_wi
...
 
-    provider_tasks.deliver_sms.apply_async.assert_called_once_with(
-        [notification_id],
-        queue="send-sms"
-    )
-
-    persisted_notification = Notification.query.filter_by(id=notification_id).one()
-    assert persisted_notification.id == notification_id
+    assert Notification.query.count() == 1
+    persisted_notification = Notification.query.all()[0]
```
Comment (servingUpAces) [inline tests/app/celery/test_tasks.py]: Ah good - will do that.

File: tests/app/celery/test_tasks.py [hunk 2]
```diff
@@ -737,24 +724,27 @@ def test_should_use_email_template_subject_placeholders(sample_email_template_wi
...
+    provider_tasks.deliver_email.apply_async.assert_called_once_with(
+        [persisted_notification.id], queue='send-email'
+    )
 
 
 def test_should_use_email_template_and_persist_without_personalisation(sample_email_template, mocker):
     notification = _notification_json(sample_email_template, "my_email@my_email.com")
     mocker.patch('app.celery.provider_tasks.deliver_email.apply_async')
 
+    assert Notification.query.count() == 0
```
Comment (servingUpAces) [inline tests/app/celery/test_tasks.py]: ok

File: app/models.py [hunk 3]
```diff
@@ -538,52 +538,31 @@ def personalisation(self, personalisation):
...
+                     job_row_number,
+                     created_at):
         return cls(
             template_id=template_id,
             template_version=template_version,
             to=recipient,
             service_id=service_id,
             status='created',
-            created_at=datetime.datetime.utcnow(),
+            created_at=created_at,
```
Comment (leohemsted) [inline app/models.py]: ah right 👍

File: app/celery/tasks.py [hunk 4]
```diff
@@ -122,27 +127,35 @@ def send_sms(self,
         return
 
     try:
-        dao_create_notification(
-            Notification.from_api_request(
-                created_at, notification, notification_id, service.id, SMS_TYPE, api_key_id, key_type
```
Comment (servingUpAces) [inline app/celery/tasks.py]: Yes it is not being used now. I think I have to create a new method without the notificaiton_id, start using the new method, deploy that, then remove the old method. I was going to worry about it in another PR.

**Pull request merged**